File: internal/state/state.go
package state

type AppState struct {
	ServerURL          string            `json:"serverUrl"`
	HideStatusBar      bool              `json:"hideStatusBar"`
	ZoomLevel          int               `json:"zoomLevel"`
	PinnedUrls         []string          `json:"pinnedUrls"`
	RecentUrls         []string          `json:"recentUrls"`
	SetupComplete      bool              `json:"setupComplete"`
	KeepScreenOn       bool              `json:"keepScreenOn"`
	Orientation        int               `json:"orientation"` // 0=자동, 1=세로, 2=가로 (데스크탑에선 창 모드 대응)
	RotationUrls       []string          `json:"rotationUrls"`
	ShowRotationBtns   bool              `json:"showRotationBtns"`
	PreloadUrls        []string          `json:"preloadUrls"`
	UrlAliases         map[string]string `json:"urlAliases"`
	ScrollLock         bool              `json:"scrollLock"`
	GridMode           bool              `json:"gridMode"`
	ManualTrustedHosts []string          `json:"manualTrustedHosts"`
	HostZoomLevels     map[string]int    `json:"hostZoomLevels"`
}

func NewDefaultState() *AppState {
	return &AppState{
		ServerURL:        "http://your-server:7800",
		HideStatusBar:    true,
		ZoomLevel:        100,
		PinnedUrls:       []string{},
		RecentUrls:       []string{},
		RotationUrls:     []string{},
		PreloadUrls:      []string{},
		UrlAliases:       make(map[string]string),
		HostZoomLevels:   make(map[string]int),
		ShowRotationBtns: true,
	}
}

File: src/shared/state.ts
export interface AppState {
  serverUrl: string
  hideStatusBar: boolean
  zoomLevel: number
  pinnedUrls: string[]
  recentUrls: string[]
  setupComplete: boolean
  keepScreenOn: boolean
  orientation: number
  rotationUrls: string[]
  showRotationBtns: boolean
  preloadUrls: string[]
  urlAliases: Record<string, string>
  scrollLock: boolean
  gridMode: boolean
  manualTrustedHosts: string[]
  hostZoomLevels: Record<string, number>
}

export function defaultState(): AppState {
  return {
    serverUrl: 'http://your-server:7800',
    hideStatusBar: true,
    zoomLevel: 100,
    pinnedUrls: [],
    recentUrls: [],
    setupComplete: false,
    keepScreenOn: false,
    orientation: 0,
    rotationUrls: [],
    showRotationBtns: true,
    preloadUrls: [],
    urlAliases: {},
    scrollLock: false,
    gridMode: false,
    manualTrustedHosts: [],
    hostZoomLevels: {}
  }
}

export type WindowMode = 'normal' | 'maximized' | 'fullscreen'

export interface JBridgeApi {
  getState(): Promise<AppState>
  switchToUrl(url: string): Promise<AppState>
  deleteUrl(url: string): Promise<AppState>
  togglePin(url: string): Promise<AppState>
  toggleRotation(url: string): Promise<AppState>
  togglePreload(url: string): Promise<AppState>
  adjustZoom(delta: number): Promise<AppState>
  setAlias(url: string, alias: string): Promise<AppState>
  toggleGridMode(): Promise<AppState>
  moveUrl(url: string, delta: number): Promise<AppState>
  addTrustedHost(host: string): Promise<AppState>
  removeTrustedHost(host: string): Promise<AppState>
  toggleScrollLock(): Promise<AppState>
  toggleStatusBar(): Promise<AppState>
  toggleKeepScreenOn(): Promise<AppState>
  toggleShowRotationBtns(): Promise<AppState>
  toggleMaximize(): Promise<boolean>
  toggleFullscreen(): Promise<boolean>
  getWindowState(): Promise<WindowMode>
  openInBrowser(url: string): Promise<void>
}
